core/templates/dev/head/pages/creator_dashboard/CreatorDashboard.js

```diff
@@ -189,6 +191,86 @@ oppia.controller('CreatorDashboard', [
     };
 
+    $scope.showCreateQuestionModal = function() {
+      var question = QuestionObjectFactory.createDefaultQuestion();
+      var topicSummaries = $scope.topicSummaries;
+      $uibModal.open({
+        templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+          '/pages/creator_dashboard/create_question_modal_directive.html'),
+        backdrop: 'static',
+        keyboard: false,
+        size: 'lg',
+        resolve: {},
+        controller: [
+          '$scope', '$uibModalInstance', function(
+              $scope, $uibModalInstance) {
+            $scope.question = question;
+            $scope.topicId = null;
+            $scope.questionStateData = $scope.question.getStateData();
+            $scope.topicSummaries = topicSummaries;
+            $scope.misconceptions = [];
+            $scope.errorMessage = null;
+
+            $scope.isValidQuestion = function() {
+              var errorMessage = $scope.question.validate([]);
+              if (!$scope.topicId) {
+                $scope.errorMessage = 'Please choose a topic before submitting';
+              } else if (errorMessage === false) {
+                $scope.errorMessage = null;
+              } else {
+                $scope.errorMessage = errorMessage;
+              }
+              return ($scope.question.validate([]) === false);
+            };
+
+            $scope.dismissModal = function() {
+              $uibModalInstance.dismiss();
+            };
+
+            $scope.createQuestion = function() {
+              var errorMessage = question.validate([]);
+              if (!$scope.topicId) {
+                $scope.errorMessage = 'Please choose a topic before submitting';
+              } else if (errorMessage === false) {
+                $scope.errorMessage = null;
+                $uibModalInstance.close({
+                  question: question,
+                  topicId: $scope.topicId
+                });
+              } else {
+                $scope.errorMessage = errorMessage;
+              }
+            };
+          }
+        ]
+      }).result.then(function(result) {
+        var topicVersion = null;
+        for (var i = 0; i < topicSummaries.length; i++) {
+          if (topicSummaries[i].id === result.topicId) {
+            topicVersion = topicSummaries[i].version;
+            break;
+          }
+        }
+        if (!topicVersion) {
+          $log.error('Unable to match topic id selected with topic choices.');
+        }
+        $http.post('/generalsuggestionhandler/', {
+          suggestion_type: 'add_question',
+          target_type: 'topic',
+          target_id: result.topicId,
+          target_version_at_submission: topicVersion,
+          change: {
+            cmd: 'create_new_fully_specified_question',
+            question_dict: result.question.toBackendDict(true),
+            skill_id: null
+          },
+          description: 'question title'
```

Comment:
What's this "description"? For the exploration feedback threads it is:

        description: str. Learner-provided description of suggestion changes.

So, if it's like a commit message, perhaps we should have something a bit more clear? Like, "Update the question title"? Or "Create new question"? Or "Initialize question with title XXX"?

Basically, I don't understand the value for this field. Please explain what it should do and pick an appropriate value for it.

Reply:
Initially I thought in the questions tab, I will show the description and the status (Similar to the feedback tab view for explorations, where the suggestion descriptions are shown to you). However, for consistency with the other questions being displayed on that page, I had used question state data in that view. So this is actually not being used! I'll remove it. 

Sorry for the confusion.

Reply:
Done